astro.config.mjs

```diff
@@ -1,4 +1,5 @@
 import { defineConfig } from "astro/config";
+import astroExpressiveCode from "astro-expressive-code";
```

Comment:
Here was the tricky part of the troubleshooting process. I realised that for this implementation, I actually did not install the MDX integration that typical Astro sites would need for MDX support (https://docs.astro.build/en/guides/markdown-content/) 🤔

My conclusion is that somehow because Starlight comes bundled with MDX, it proliferated throughout the entire site? Might be the case (see https://discord.com/channels/830184174198718474/1242248518433898618/1243107372474568735)

Starlight uses Expressive Code for rendering code blocks (see https://starlight.astro.build/guides/authoring-content/#code-blocks), and for our other documentation sites, the expressive code configuration is part of the Starlight integration config.

However, because our Engineering blog is done with Astro's content collections, we have to add `astro-expressive-code` in order to configure it properly.